src/main/java/io/lettuce/core/RedisHandshake.java

```diff
@@ -281,6 +270,25 @@ private CompletableFuture<Void> applyPostHandshake(Channel channel, String redis
         return dispatch(channel, postHandshake);
     }
 
+    private CompletableFuture<Void> applyFireAndForget(Channel channel) {
```

Comment:
to be fair, the method name doesn't reflect what is happening here as lenient error handling happens in the calling code.

Looking at the code, it might make sense to introduce stronger guards to avoid `ArrayList` creation if neither library name nor library version are set and so avoid also the call to `dispatch`.

Just a couple of thoughts.

Reply:
Fair assessment :) Was a bit hasty with the change, will update with your suggestions. Thanks!

Reply:
TBH, I'm not sure about fire+forget. 

There are quite some commands being invoked on startup and any failures can go unnoticed. Someone relying on the client name who has renamed the command doesn't find out about a failure until they investigate the client name on their own.

If someone encounters failures because of these commands, then they should not set the client name, library version, ….

What is the rationale to ignore errors?

Reply:
Could you give me an example where this metadata is critical to the operation of the client?
So far I assumed this is mostly tracing information. Are there flows where one client expects to find another one based on the name the other client provided? How would they handle this in older versions in Redis? I would be very hesitant to endorse the usage of these fields in this way

Even so I think the same rationale does not exist for lib-name and lib-version - these are definitely tracing information that is secondary to the operation of the driver.

Here is an excerpt from the documentation  for the [CLIENT SETINFO](https://redis.io/docs/latest/commands/client-setinfo/):

```
Client libraries are expected to pipeline this command after authentication on all connections and ignore failures since they could be connected to an older version that doesn't support them.
```

As I mentioned different flavors of the Redis server handle this in different versions and to embed a support matrix in the client would be counterproductive when the operation can simply fail. #2817 is a testament to this issue.

Would you consider it an improvement if we log the issue?

Reply:
@mp911de I am just adding my 2 cents here. Command renaming is seldom used nowadays because Redis provides more efficient ways to secure the connection, etc.

Reply:
Probably logging at the debug level is a good common ground. That being said, `CLIENT SETNAME` belongs into the same category.

Reply:
Fair, let me prepare another change. Thanks for considering this!

Reply:
@mp911de what about this version?